docs/book_examples/src/data_md.rs

```diff
@@ -21,7 +21,7 @@ struct TodoItem {
     due_date: Option<Arc<DateTime>>,
     // You can specify a custom comparison fn
     // (anything with the signature (&T, &T) -> bool).
-    #[data(same_fn = "PartialEq::eq")]
+    #[data(eq)]
```

Comment:
Looking at this now, I'm not *totally* sure it's an improvement, or at least I'm not totally sure about this name. I think reading the code, it is very obvious what the old version does (especially since this sort of annotation is used in things like serde) but the second version looks very cryptic.

Reply:
I was also confused by naming, I like `#[data(partial_eq)` more. But decided to go with the original naming.

I think this is a common case, We should do _something_ about it. I am not sure if this is the best thing.
`#[serde(default)]` and `#[serde(default = "..")]` is similar to this.

and yes this comment needs to be updated

Reply:
Okay, I find that serde example compelling.


`#[data(partial_eq)]` or `#[data(partialeq)]`? `#[data(eq)]`? `#[data(equality)]`? `#[data(equals)]`? `#[data(==)]`? No, that's horrible.

so... `#[data(partial_eq)]` or `#[data(eq)]`, I guess.

An advantage of `eq` is that once you know it exists it's harder to get wrong. So maybe it's better, after all? But I'm okay either way.

I agree that the docs should show examples of all attributes.